### ISSUE TITLE
how to add lstm layer as meta learner

### ISSUE THREAD

**Issue by realize-qzq** [AntreasAntoniou/HowToTrainYourMAMLPytorch#18]
Most of maml code use cnn and use nn.functional to realize it. But sometimes we need lstm or other module which doesn't appear in nn.functional. I try to load net parameters repeatly in inner loop but the accuracy doesn't raise.

**Comment by AntreasAntoniou** [comment]
You would need to build an LSTM in the same flavour as the 'Meta' layers. This can be achieved by diving into the nn.LSTM layer and creating a new layer which has that functionality + can receive parameters to use for its internal functions.

**Comment by tk1363704** [comment]
> You would need to build an LSTM in the same flavour as the 'Meta' layers. This can be achieved by diving into the nn.LSTM layer and creating a new layer which has that functionality + can receive parameters to use for its internal functions.

I am faced with the same problem but I can not understand what is 'diving into the nn.LSTM layer and creating a new layer which has that functionality + can receive parameters to use for its internal functions'.
If possible, could you please explain it more deeply, if it's not too much trouble. Cheers!
